### ISSUE TITLE
[Feature Request] Virtual Path Editing

### ISSUE THREAD

**Issue by catkid15368** [d0k3/GodMode9#394]
I would like a feature where you could use hex editor to edit a CIA in virtual path.
e.x.
When you mount a .cia file and you go to the icon. I wish you could use the hex editor.

**Comment by SirNapkin1334** [comment]
You can dump the CIA, mount it, and grab all the mounted stuff. Then edit, and recompile the CIA on your computer.

**Comment by catkid15368** [comment]
The thing is it takes over almost 35 minutes to copy. Also I don't know how to compile a CIA file. Can I use GM9?

**Comment by ihaveamac** [comment]
These formats are not designed to be edited, so the only way is to rebuild the cia and the files inside.

**Comment by catkid15368** [comment]
How do I rebuild the CIA. Cuz if I need a windows computer to do that... then I can't.

**Comment by ihaveamac** [comment]
You need a PC, not Windows specifically. You can use 3dstool and makerom to rebuild parts of the NCCH and CIA.

**Comment by SirNapkin1334** [comment]
You can do it with makerom / 3dstool, which can be used on a Mac.

**Comment by catkid15368** [comment]
I only have a chromebook

**Comment by SirNapkin1334** [comment]
Dual-booting it to run Linux is probably one of your only options.

**Comment by catkid15368** [comment]
how do i do that?

**Comment by catkid15368** [comment]
If it requires dev mode. No thanks.

**Comment by ihaveamac** [comment]
Well since this feature is basically impossible you should close the issue.

**Comment by d0k3** [comment]
Eh, @catkid15368 - this is not a support forum, if you need further help you should ask elsewhere.

Regarding your feature request - changing *anything* inside a CIA requires a full recompile of said CIA. With the crappy 3DS card reader / writer hardware, this will take something between 10...15 minutes. Copying it to a computer, using a half decent USB sdcard reader or even Wifi and FTP won't take that long.

So, in short - use a PC. There is no practical way of doing this in GM9.
